Tests/ApolloCodegenTests/FileManagerExtensionsTests.swift

```diff
@@ -24,643 +24,714 @@ class FileManagerExtensionTests: XCTestCase {
   }
 
   func test_deleteDirectory_givenFileDoesNotExistAndIsNotDirectory_shouldSucceed() throws {
     // given
-    let mocked = MockFileManager(fileExists: { (path: String, isDirectory: UnsafeMutablePointer<ObjCBool>?) in
+    let mocked = MockFileManager()
+
+    mocked.set(closure: .fileExists({ path, isDirectory in
       expect(path).to(equal(self.uniquePath))
       expect(isDirectory).notTo(beNil())
 
       isDirectory?.pointee = false
       return false
-    })
+    }))
 
     // then
     expect(try mocked.apollo.deleteDirectory(atPath: self.uniquePath)).notTo(throwError())
-    expect(mocked.blocksCalled).to(equal([.fileExists]))
+    expect(mocked.allClosuresCalled).to(beTrue())
   }
 
   func test_createFile_givenContainingDirectoryDoesExistAndFileCreated_shouldNotThrow() throws {
     // given
     let parentPath = URL(fileURLWithPath: self.uniquePath).deletingLastPathComponent().path
-    let mocked = MockFileManager(fileExists: { (path: String, isDirectory: UnsafeMutablePointer<ObjCBool>?) in
+    let mocked = MockFileManager()
+
+    mocked.set(closure: .fileExists({ path, isDirectory in
       expect(path).to(equal(parentPath))
       expect(isDirectory).notTo(beNil())
 
       isDirectory?.pointee = true
       return true
 
-    }, createFile: { (path: String, data: Data?, attr: FileAttributes?) in
+    }))
+    mocked.set(closure: .createFile({ path, data, attr in
       expect(path).to(equal(self.uniquePath))
       expect(data).to(equal(self.uniqueData))
       expect(attr).to(beNil())
 
       return true
 
-    }, createDirectory: { (path: String, createIntermediates: Bool, attr: FileAttributes?) in
-      fail("The containing directory already exists, createDirectory should not be called.")
-    })
+    }))
 
     // then
     expect(
       try mocked.apollo.createFile(atPath: self.uniquePath, data:self.uniqueData)
     ).notTo(throwError())
-    expect(mocked.blocksCalled).to(equal([.fileExists, .createFile]))
+    expect(mocked.allClosuresCalled).to(beTrue())
   }
 
   func test_createFile_givenContainingDirectoryDoesExistAndFileNotCreated_shouldThrow() throws {
     // given
     let parentPath = URL(fileURLWithPath: self.uniquePath).deletingLastPathComponent().path
-    let mocked = MockFileManager(fileExists: { (path: String, isDirectory: UnsafeMutablePointer<ObjCBool>?) in
+    let mocked = MockFileManager()
+
+    mocked.set(closure: .fileExists({ path, isDirectory in
       expect(path).to(equal(parentPath))
       expect(isDirectory).notTo(beNil())
 
       isDirectory?.pointee = true
       return true
 
-    }, createFile: { (path: String, data: Data?, attr: FileAttributes?) in
+    }))
+    mocked.set(closure: .createFile({ path, data, attr in
       expect(path).to(equal(self.uniquePath))
       expect(data).to(equal(self.uniqueData))
       expect(attr).to(beNil())
 
       return false
 
-    }, createDirectory: { (path: String, createIntermediates: Bool, attr: FileAttributes?) in
```

Comment:
The 'strict' nature of `MockFileManager` will fail tests calling functions that aren't stubbed, i.e.: don't need fail blocks like this.